Sunjom

sunjom opened the pull request:
inline-block , grid , flex 기능 구현했습니다.

Commit:
[FE] Feat : inline-block 기능 구현

Commit:
[FE] Feat : grid 기능 구현

Commit:
[FE] Feat : flex 기능 구현

ccconac commented on mission/1주차/src/html/inline-block.html:
```diff
@@ -5,10 +5,63 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>inline-block layout</title>
     <link rel="stylesheet" href="../css/inline-block.css" />
+    <style>
```
HTML 내에 스타일 코드를 넣는 것을 `Internal Style Sheet`라고 하는데요. 다른 HTML 파일에서 해당 css와 거의 유사해 재사용하고 싶은 경우 재사용을 할 수 없다는 단점이 있어서 이보다는 별도의 css 파일을 만들어 HTML 문서와 연결하는 방법인 `Linking Style Sheet`을 권장합니다. 🧐

처음에 제공한 css directory 내의 inline-block.css 파일에서 진행했다면 좋았을 것 같아요!

ccconac commented on mission/1주차/src/html/inline-block.html:
```diff
@@ -5,10 +5,63 @@
     <title>inline-block layout</title>
     <link rel="stylesheet" href="../css/inline-block.css" />
+    <style>
+      .width{
```
속성의 이름과 동일하고, 의미를 명확하게 드러낼 수 없는 네이밍은 지양하는 것이 좋습니다! :> 

내가 원했던 기능이 무엇인지를 생각하고 그걸 이름에 담아 낼 수 있다면 좋을 것 같아요. 👀

ccconac commented on mission/1주차/src/html/inline-block.html:
```diff
@@ -5,10 +5,63 @@
+        background-color:#dff3df;
+        text-align: center;
+        position:relative;
+        bottom:10%;
```
`inline-block`이나 `text-align`이 반복되고 있는데, 이런 경우는 아래처럼 작성하면 해당 스타일을 필요로 하는 모든 클래스에 적용이 가능해요!

```css
.header,
.left-side,
.main,
.right-side,
.footer {
  display: inline-block;
}
```

그리고 모든 클래스에서 속성의 의미를 제대로 알지 못하고 무작위로 사용하는 것처럼 보이는 부분들이 많은데, 해당 속성이 어떤 역할을 하는지 찾아보고 사용하면 좋을 것 같아 보여요. 이런 것들이 많아지면 추후에 의도가 명확하지 않은 스타일들이 중첩되어서 원하는 레이아웃과 거리가 멀어질 수 있어요. 🧐

특히 `position`에 대한 학습이 조금 더 필요해 보입니다. 해당 피드백은 아래에 있는 모든 코드들에도 해당해요! 👀

ccconac commented on mission/1주차/src/html/inline-block.html:
```diff
@@ -5,10 +5,63 @@
+      </div>
+      <div class="center">
+        <div>center</div>
+      </div>
```
레이아웃을 위한 DOM 구조에 대해 조금 더 고민해 보면 좋을 것 같아요. 🤔

또한, 이미 생성되어 있는 스타일 클래스가 있음에도 불구하고 `Inline Style Sheet`로 스타일을 적용하는 부분들이 있는데 이런 스타일들은 클래스 내에 포함시켜 주세요!

**현재 코드**
```html
<div class="width" style="top: 80%;">
    <div class="width_text_center">footer</div>
</div>
```

**수정 코드**
```html
<div class="width">
    <div class="width_text_center">footer</div>
</div>
```

```css
.width {
  display: inline-block;
  position:relative;

  width: 100%;
  height:10%;

  top: 80%;
  background-color: #5ed7ff;
}
```

ccconac commented on mission/1주차/src/html/flex.html:
```diff
@@ -6,9 +6,48 @@
     <link rel="stylesheet" href="../css/flex.css" />
   </head>
+  <style>
+    .main_Side{
```
클래스 및 아이디 네이밍 시, 언더바로 글자를 구분할 경우 모든 글자는 소문자로 구성되게 해 주세요.

```css
.main_side {}
```

또한 메인을 이루는 container로 보여서, 아래 예시처럼 작명해 주면 조금 더 이름에 명확하게 의미를 드러낼 수 있을 것 같아요. :>


```css
.main_container {}
```

ccconac commented on mission/1주차/src/html/flex.html:
```diff
@@ -6,9 +6,48 @@
+      display: flex;
+      width: 100%;
+      height: 100%;
+      justify-content: space-between;
```
해당 방법도 좋지만, `flex-direction: row`를 사용해 주면 `space-between`을 사용해 간격을 만들어 주지 않아도 자동으로 가로로 정렬 가능합니다. 👀

ccconac commented on mission/1주차/src/html/flex.html:
```diff
@@ -6,9 +6,48 @@
+    .Side{
+      width: 10%;
+      height: 100%;
+      background-color:#dff3df;
```
요구 사항 README를 조금 더 꼼꼼하게 확인해 주시면 좋을 것 같아요. :> 요구 사항에서 제시한 색상 코드는 `#dff3fd`입니다!

ccconac commented on mission/1주차/src/html/flex.html:
```diff
@@ -6,9 +6,48 @@
+      height: 100%;
+      justify-content: space-between;
+    }
+    .main_Top{
```
해당 클래스명도 `.main_top`보다는 `.header_container`가 좋아 보입니다. 굳이 필요한 클래스로 보이지는 않지만요. 🤔

ccconac commented on mission/1주차/src/html/flex.html:
```diff
@@ -6,9 +6,48 @@
+      height: 100%;
+      background-color:#dff3df;
+    }
+    .Top{
```
`.Top` → `.header`

ccconac commented on mission/1주차/src/html/flex.html:
```diff
@@ -6,9 +6,48 @@
+      </div>
+      <div class="main_Side">
+        <div class="Side text">left</div>
+        <div class="text">mid</div>
```
요구 사항 README를 조금 더 꼼꼼하게 확인해 주시면 좋을 것 같아요!

ccconac commented on mission/1주차/src/html/grid.html:
```diff
@@ -5,10 +5,52 @@
+      .grid{
+        display : grid;
+        grid-template-columns: repeat(6,1fr);
+        grid-template-rows:repeat(4,162px);
```
`grid`에 대한 숙지가 더 필요해 보여요. 🧐

레이아웃의 행을 총 몇 개로 나누면 좋을지, 열은 몇 개로 나누면 좋을지 큰 틀을 잡아 우선적으로 생각해 본 뒤에 다시 코드를 작성해 보면 좋을 것 같습니다.

ccconac reviewed (commented):
1주차 미션 수행해 주신 점 감사합니다! 수고 많으셨어요. 🎉 

커밋 컨벤션을 지켜서 기능을 분리해 커밋해 주신 점 너무 좋았어요. 👍👍

전체적으로 아직 css에 대한 개념과 이해도가 많이 부족해 보여서, 해당 미션에 대한 학습을 스스로 조금 더 진행해 보시는 게 좋을 것 같다고 생각돼요. 👀 레이아웃은 말을 굳이 할 필요가 없을 정도로 기초적인 부분이다 보니, 이 부분이 숙지가 안 되면 진행이 불가능하다고 생각합니다. 

우선 다시 제가 말씀드린 내용을 중점으로 수정해 보시고 도저히 안 되겠다 하시면 solution branch 1주차 답안 참고해서 진행해 주세요. :) 

> 단, 제 답은 그냥 참고용일 뿐이지 그 외의 여러 방법으로 구현 가능합니다. 🚨

또한, 요구 사항과 다르게 수행된 부분들이 많이 보여요. 조금 더 꼼꼼하게 요구 사항을 읽어 주시면 좋을 것 같습니다! 👀

공부에 참고가 될 만한 내용들이 담긴 링크를 아래에 걸어 둡니다.

**Grid**
> https://studiomeal.com/archives/533 

**Flex**
> https://studiomeal.com/archives/197

Commit:
[FE] Refactor : inline-block 소스코드 수정

코드명 변경
Linking Style Sheet 사용
반복 속성 제거
클래스 이름 소문자로 변경
요구사항 내용 확인 후 변경

Commit:
[FE] Refactor : grid  소스코드 수정

반복 속성 제거
클래스 이름 소문자로 변경
요구사항 내용 확인 후 변경
grid-template-areas 속성 사용
text_center를 위해 display:grid 사용

Commit:
[FE] Refactor : flex  소스코드 수정
